src/main/java/core/basesyntax/Pair.java

```diff
@@ -41,10 +41,14 @@ public boolean equals(Object obj) {
         if (obj == null) {
             return false;
         }
-        Pair pair = (Pair) obj;
+        Pair<F, S> pair = (Pair) obj;
         if (pair.getClass().equals(this.getClass())) {
-            return (first == pair.getFirst() || first.equals(pair.getFirst()))
-                    && (second == pair.getSecond() || second.equals(pair.getSecond()));
+            return (first == pair.getFirst()
+                    || first != null && pair.getFirst() != null
```

Comment:
```suggestion
                    || first != null 
```
ця перевірка надлишкова, адже ми викликаємо equals на first, отже тільки first не повинен бути null